add button to sort by created_by; button to flip the sort direction

artlu99 opened the pull request:
- sort by id or created_by
- sort can be ascending or descending

artlu99 committed:
add button to sort by created_by; button to flip the sort direction

davidfurlong commented:
<img width="476" alt="Screenshot 2024-01-27 at 19 38 01" src="https://github.com/davidfurlong/farcaster-channels/assets/614768/4233729f-4e8d-4e49-9f6b-69d473fcbc5b">

I like the idea of adding this functionality and thanks for the PR! I think the UI needs a bit of work before merging; want to keep it simple and clean
- sort buttons are misaligned
- sort buttons are confusing and icons aren't explanatory enough
- Showing date created on the channel is adding lots of visual clutter

Would suggest the following changes:
- Sort dropdown with four options (Alphabetical, A->Z), (Alphabetical, Z->A), (Date, Newest first), (Date, Oldest First)
- Make sure the dropdown is aligned and fits on mobile + web
- Remove the date from the channel item, instead adding it to the Channel modal detail view

artlu99 committed:
replace sort buttons with dropdown menu; move created time detail

artlu99 committed:
npm package-lock.json

artlu99 committed:
TEMP commit - REVERT BEFORE MERGE

point Github link to fork, not original

artlu99 committed:
fix dropdown behavior

artlu99 committed:
move sort button by 1 pixel

artlu99 committed:
fix sort by newest

artlu99 committed:
bigger sort button for mobile

artlu99 committed:
revert TEMP commit

artlu99 commented:
Request review of changes:
- dropdown menu with single icon
- mobile buttons are larger
- removed date from channel item, added it to the Channel modal detail view

Preview deployment is (temporarily) at farcaster-channels.artlu.xyz

davidfurlong commented on app/results.tsx:
```diff
@@ -69,10 +81,29 @@ export function Results() {
+          (a.id > b.id ? 1 : -1) * Math.sign(sortMethodObj.direction)
+        );
+
+    setFilteredResults(sortedResults.filter((result) => {
```
I think it makes sense to filter and then sort, for performance

davidfurlong replied:
Doing this via setstate and not just a var like previously is increasing input lag somehow

artlu99 replied:
I agree about the input lag, which appears to get worse with more rows. The commit below attempts to limit the number of rows rendered, by using @tanstack/react-table

davidfurlong commented on app/results.tsx:
```diff
@@ -207,52 +239,65 @@ export function Results() {
+            type="search"
+            value={q}
+            onChange={(e) => setQ(e.target.value)}
+            className="w-[400px] ring-violet-500 focus:ring-1 outline-none max-w-full bg-violet-50 border border-violet-200 text-violet-900 text-sm rounded focus:border-violet-300 block p-2 dark:bg-violet-950 dark:border-violet-600 dark:placeholder-violet-400 dark:text-violet-300"
```
can we make this have className `cursor-pointer`?

Also is there a reason you're using a `Dropdown` here instead of a `Select`? seems more fitting to use a Select.

davidfurlong reviewed (changes requested):
Hey, thanks for the contribution - it's almost there!

artlu99 committed:
fix laggy rendering by using @tanstack/react-table

artlu99 committed:
add cursor-pointer on search field

artlu99 commented:
Request review of changes:

- react-table replaces a full, raw table with 4000+ rows
- component includes filtering, sorting and pagination 

Preview deployment is (temporarily) at farcaster-channels.artlu.xyz

artlu99 committed:
better spacing for mobile

davidfurlong commented:
<img width="1410" alt="Screenshot 2024-02-26 at 07 26 42" src="https://github.com/davidfurlong/farcaster-channels/assets/614768/9f279815-b150-4947-99f5-4d2255d9389f">

Almost there!
I think the table change has made some of the UX significantly worse, namely scrolling to discover. I think infinite scroll is greatly superior to pagination, and these table columns are oddly spaced (image should have correct fixed width) and the table padding is awkward and unncessary. The loading state currently shows "No results"

artlu99 committed:
test performance with larger page sizes

artlu99 committed:
Revert "test performance with larger page sizes"

This reverts commit 1aa755980931b1debd382c28673309e61d878a66.

artlu99 commented:
Closing this PR without merging. Will continue to work in the fork until the UX issues are resolved
